### PR TITLE
CMS-211 Code review

### PR DESCRIPTION
**Pull request by sdonk**

### CODE REVIEW

File: pir_client/__init__.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+from functools import lru_cache
```
Comment (sdonk) [inline pir_client/__init__.py]: this code should not be in the module `__init__.py` file

File: pir_client/__init__.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+from directory_client_core.base import BaseAPIClient
+
+
+class PIRAPIClient(BaseAPIClient):
```
Comment (sdonk) [inline pir_client/__init__.py]: Endpoints should be a list class attribute, https://github.com/uktrade/directory-api-client/blob/master/directory_api_client/company.py#L9

File: pir_client/__init__.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                'pdf': 'https://pir-invest.s3.amazonaws.com/media/usatechRollo2018-05-25.pdf',
+                'date_created': '2018-05-25T11:23:45.406400Z'
+            }
+
+        """
+        options = self.get_options()
+
+        if 'market' in options:
+            market_choices = [
+                o['value'] for o in options['market']['choices']
```
Comment (sdonk) [inline pir_client/__init__.py]: single letter variable names are not meaningful

File: pir_client/__init__.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        if 'sector' in options:
+            sector_choices = [
+                o['value'] for o in options['sector']['choices']
+            ]
+        else:
+            sector_choices = []
+
+
+        if 'market' in data and 'country' in data:
+            raise ValueError('Cannot provide country and market')
```
Comment (sdonk) [inline pir_client/__init__.py]: client should raise its own exception that needs to be caught in the code using the client

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+-e git+https://git@github.com/uktrade/directory-client-core.git@2f366ea0a1e713165ced30cd678359ee67d2ed39#egg=directory_client_core
```
Comment (sdonk) [inline requirements.txt]: This should come from `pypi`https://github.com/uktrade/directory-api-client/blob/master/requirements.txt#L3

File: pir_client/tests/test_client.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
+import httpretty
```
Comment (sdonk) [inline pir_client/tests/test_client.py]: To keep this client library consistent with the others we have a `DummyClient` needs to be defined and `httpretty` should be removed https://github.com/uktrade/directory-api-client/blob/master/directory_api_client/dummy_client.py

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+def get_requirements():
+    return [str(r.req) for r in list(parse_requirements(
+        'requirements.txt',
+        session=pip.download.PipSession()
+    ))]
+
+
+setup(
+    name='pir_client',
+    version="0.0.5",
```
Comment (sdonk) [inline setup.py]: This needs to be in a version file https://github.com/uktrade/directory-api-client/blob/master/directory_api_client/version.py and https://github.com/uktrade/directory-api-client/blob/master/setup.py

**Comment by sdonk** [comment]
This code is lacking CI and test coverage integration. Use https://github.com/uktrade/directory-api-client for reference

File: pir_client/__init__.py (hunk 2 above)
Comment (eadpearce) [inline pir_client/__init__.py]: use `__init__.py` for package config not for the main bulk of the code https://docs.python.org/3/tutorial/modules.html#packages
check directory-cms-client or directory-api-client for how to structure https://github.com/uktrade/directory-cms-client https://github.com/uktrade/directory-api-client

**Pull request merged**